apps/trackers/service.py

```diff
@@ -0,0 +1,105 @@
+"""
+Task Manager API endpoints
+"""
+import logging
+
+from django.db import transaction
+from jira.exceptions import JIRAError
+from rest_framework.response import Response
+
+from collectors.jiraffe.core import JiraQuerier
+from osidb.models import Affect, PsUpdateStream, Tracker
+
+from .bts_tracker import JiraTracker
+from .constants import JIRA_SERVER
+
+logger = logging.getLogger(__name__)
+
+
+class JiraTrackerQuerier(JiraQuerier):
+    """
+    Jira query handler for tracker management.
+    This class encapsulates calls for Jira doing validations
+    and it methods return data requested with HTTP status code
+    """
+
+    def __init__(self, token) -> None:
+        """
+        Instantiate a new JiraTrackerQuerier object.
+
+        Keyword arguments:
+        token -- user token used in every request to Jira
+        """
+        self._jira_server = JIRA_SERVER
+        self._jira_token = token
+
+    def get_bts_tracker(self, bts_id):
+        """get Jira tracker given its string key or integer id"""
+        try:
+            return Response(data=self.jira_conn.issue(bts_id).raw, status=200)
+        except JIRAError as e:
+            return Response(data=e.response.json(), status=e.status_code)
+
+    def create_bts_affect_tracker(
+        self, affect: Affect, tracked_streams: list[PsUpdateStream]
+    ):
+        """
+        Generates a tracker in BTS given an Affect.
+        If multiples streams is given creates one tracker per stream.
+        """
+        if affect.trackers.exists():
+            return Response(data="Affect already have trackers created.", status=409)
```

Comment:
I am not sure that this check is correct. We can file different trackers for a specific affect in different times. There is no existing restriction on this. Having some trackers already should not be a blocker for creating new ones.

Of course there should be only a single tracker for each stream so we should restrict this.

Reply:
changed validation during the tracker creation in order to limit 1 tracker per stream for each affect.